dutree/dutree.py

```diff
@@ -468,34 +503,46 @@ def human(value):
 
 
 def main():
-    pathname = None
-    if len(sys.argv) == 2:
-        def getsize(node):
-            return node.app_size()
-
-        pathname = sys.argv[1]
-        use_apparent_size = True
+    parser = argparse.ArgumentParser(prog='dutree',
+                                     description='dutree shows a summary of \
+                                        the directories/files which take up\
+                                        the most space.')
+    parser.add_argument('--count-blocks', help='count block size not app size',
+                        action='store_true')
+    parser.add_argument('--xdev', help='stay on the same fileystem',
+                        action='store_true')
+    parser.add_argument('--no-skip-proc-sys',
+                        help='do not skip the filesystem devices\
+                            for proc and sys directories,\
+                            by default these are skipped',
+                        action='store_true')
+    parser.add_argument('path')
+
+    args = parser.parse_args()
+
+    use_apparent_size = True
+
+    if args.count_blocks:
+        use_apparent_size = False
 
-    elif len(sys.argv) == 3:
         def getsize(node):
             return node.use_size()
+    else:
 
-        if sys.argv[1] == '--count-blocks':
-            pathname = sys.argv[2]
-        elif sys.argv[2] == '--count-blocks':
-            pathname = sys.argv[1]
-        use_apparent_size = False
-
-    if pathname is None:
-        sys.stderr.write('Usage: dutree [--count-blocks] PATH\n')
-        sys.exit(1)
+        def getsize(node):
+            return node.app_size()
 
-    run(pathname, use_apparent_size, getsize)
+    run(args.path, use_apparent_size, getsize, args.xdev,
+        (not args.no_skip_proc_sys))
```

Comment:
Now we have so many arguments (> 3 or 4) here that it starts to become necessary to name them.

I would swap the getsize and use_apparent_arguments in the run-function and then go with:
```python
    run(args.path, getsize, use_apparent_size=(not args.count_blocks),
        xdev=args.xdev, skip_proc_sys=(not args.no_skip_proc_sys))
# ...

def run(pathname, getsize, use_apparent_size, xdev, skip_proc_sys):
```

Reply:
changed